src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -184,9 +182,24 @@ public function findMany($ids, $columns = ['*'])
             return $this->model->newCollection();
         }
 
-        $this->query->whereIn($this->model->getQualifiedKeyName(), $ids);
+        return $this->byKey($ids)->get($columns);
+    }
+
+    /**
+     * Add where clause with key condition to the query.
+     *
+     * @param  mixed  $id
```

Comment:
What do you mean? $id can be array or int, mixed used here same as for `find` method